src/waveform/waveformwidgetfactory.cpp

```diff
@@ -928,161 +924,107 @@ void WaveformWidgetFactory::evaluateWidgets() {
             category = WaveformT::category();
         };
 
-        switch(type) {
-        case WaveformWidgetType::EmptyWaveform:
-            setWaveformVarsByType.operator()<EmptyWaveformWidget>();
-            break;
-        case WaveformWidgetType::SoftwareSimpleWaveform:
-            continue; // //TODO(vrince):
-        case WaveformWidgetType::SoftwareWaveform:
+        auto info = WaveformWidgetInfoBase::find(static_cast<WaveformWidgetType::Type>(type));
+        if (info) {
+            // waveform widget types registered with WaveformWidgetInfo's
+            widgetName = buildWidgetDisplayNameFromInfo(*info);
+            useOpenGl = info->m_useGL;
+            useOpenGles = info->m_useGLES;
+            useOpenGLShaders = info->m_useGLSL;
+            category = info->m_category;
+        } else {
+            switch (type) {
```

Comment:
once WaveformWidgetInfo's have been implemented for all WaveformWidgetTypes this switch can be removed.